src/main/webapp/profileScript.js

```diff
@@ -107,13 +108,37 @@ function addRoute(newRoute) {
   routeRating.classList.add("rating");
 
   routeDetails.innerHTML = newRoute["routeName"];
-  let emptyStart = '<span class="far fa-star"></span>';
-  routeRating.innerHTML = emptyStart.repeat(5);
+  let emptyStar = '<span class="far fa-star"></span>';
+  let halfStar = '<span class="fas fa-star-half-alt"></span>';
+  let fullStar = '<span class="fas fa-star checked"></span>';
+
+  let ratingCopy = newRoute["rating"];
+  let numberOfFullStars = 0;
+  while (ratingCopy >= 1) {
```

Comment:
I think using Math.floor should make this logic simpler to read. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/floor